core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -610,11 +701,38 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
    * directory, `None` is returned, indicating that there isn't an event log at that location.
    */
   private def getModificationTime(fsEntry: FileStatus): Option[Long] = {
+    getStatusAttribute(fsEntry, (f => f.getModificationTime()))
+  }
+
+  /**
+   * Get the size of the log, or `None` if there isn't one in the child
+   * directory of a legacy log entry
+   * @param fsEntry file status of a path
+   * @return the log size
+   */
+  private def getLogSize(fsEntry: FileStatus): Option[Long] = {
+    if (isLegacyLogDirectory(fsEntry)) None else Some(fsEntry.getLen)
+//    getStatusAttribute(fsEntry, (f => f.getLen()))
```

Comment:
-gone to the simpler version and reverted the generic "get status attribute" back into `getModificationTime()` for a smaller diff and simpler code. Its unlikely that people will be running new legacy-log applications, and if they are, well, no incremental playback for them.